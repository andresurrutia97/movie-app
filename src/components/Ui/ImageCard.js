import React from 'react';
import PropTypes from 'prop-types';

const ImageCard = ({ imgUrl }) => (
  <div
    className="custom-poster-card rounded-lg shadow-lg"
    style={{ backgroundImage: `url('${imgUrl}')` }}
  />
);

ImageCard.propTypes = {
  imgUrl: PropTypes.string.isRequired,
};

export default ImageCard;
