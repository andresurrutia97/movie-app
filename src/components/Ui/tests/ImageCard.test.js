import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ImageCard from '../ImageCard';

describe('<ImageCard/>', () => {
  const component = <ImageCard imgUrl="path" />;

  it('should render correcty ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain a backgroundImage url equal to "path" ', () => {
    const wrapper = shallow(component);
    expect(wrapper.props().style.backgroundImage).toMatch(new RegExp('path'));
  });
});
