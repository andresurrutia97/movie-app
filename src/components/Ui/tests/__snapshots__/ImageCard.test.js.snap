// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageCard/> should render correcty  1`] = `
<div
  className="custom-poster-card rounded-lg shadow-lg"
  style={
    Object {
      "backgroundImage": "url('path')",
    }
  }
/>
`;
